File: day1/part1.py
import sys

def main():
  measurements = [int(m.strip()) for m in sys.stdin.readlines()]
  
  prev = None
  increases = 0
  for m in measurements:
    if prev != None:
      if m > prev:
        increases += 1
    prev = m

  print(f"{increases}")



if __name__ == "__main__":
  main()

File: c_solutions/day1/part1.c
#include <stdio.h>

int main(void)
{
  int increases = 0;
  int prev = -1;
  while (1)
  {
    int n;
    scanf("%d\n", &n);
    if (n == -1)
      break;

    if (n > prev && prev != -1)
      increases++;

    prev = n;
  }

  printf("%d\n", increases);

  return 0;
}